docs/maintainers/code_coverage.md

```diff
@@ -0,0 +1,42 @@
+# Code Covergae
+
+1. [Overview](#overview)
+2. [Details](#details)
+
+## Overview
+
+Code coverage is important for ensuring our integrated testing covers
+the functionally of the code. The common project CI is set to reject
+pull requests which cause the overall code coverage to fall. Therefor
+it is important that any new functionality comes with tests to
+exercise it.
+
+## Details
+
+The code coverage is controlled by a file
+`coverage_config_x86_64.json` which usually sits in the root directory
+of the repository. The file controls which files are excluded from
+coverage calculations as well as the current coverage score. When the
+CI is run the new score is compared to the stored score and if it
+drops the CI run will fail. While you could include a change to the
+file in your PR to lower the coverage score this would be grounds to
+reject the pull request. A repository bot will periodically submit
+pull requests to update the coverage and ratchet up the limit.
```

Comment:
The bot still needs to be implemented. And we also need to make the coverage test optional. We can do 2 things:
- block this PR until we actually get the test optional (and do not mention the bot for now as that might take a bit more time)
- remove the parts that are mentioning the bot, merge this PR and work in parallel on making the coverage optional.

What is your preferred approach? Or do you have something else in mind?

Reply:
Did we want to make the coverage test fully optional? I thought the idea was to change the precision so a small drop wouldn't automatically fail but still keep the general principle of improving coverage. I can drop the bot language for now (or mention it in a future tense?).

Reply:
Hmm, actually yes, it should be updated from time to time because otherwise the difference might increase, but it is no longer so important because we're not allowing huge differences.

Reply:
I did some pseudo science on the kvm-ioctls PRs that needed coverage update. Out of 19 PRs:
- 10 had a coverage difference of 0.1
- 5 had a coverage difference of 0.2
- 1 had a coverage difference of 0.3
- 3 had a coverage difference which was more or equal to 0.6

So 84% of PRs had a difference of less or equal to 0.3. To account for some errors here and there, I would just make any difference less than 0.5 not fail the coverage test.

Reply:
I've dropped the mention of the bot and tweaked a bit of the wording regarding actions depending on if coverage has increased or decreased.